### ISSUE TITLE
`Class not found when unmarshalling: com.google.android.gms.common.api.Scope`

### ISSUE THREAD

**Issue by nabati** [StasDoskalenko/react-native-google-fit#211]
As far as I can tell, I've followed all the instructions on setting up the project, creating the Oauth Client IDs etc etc, but when trying to to authorize with Google Fit, I first get the regular "account menu" where I can select one of my accounts on the device. 

Once I have selected the account, the modal shifts size and shows a spinner that just keeps on spinning indefinitely. 

Checking `logcat` gives this output. Any indication on how to proceed?

```
02-07 21:53:36.234 23797 23797 I RNGoogleFit: Authorization - Failed Authorization Mgr:ConnectionResult{statusCode=SIGN_IN_REQUIRED, resolution=PendingIntent{3ca92d7: android.os.BinderProxy@f4837c4}, message=null}
02-07 21:53:36.237   977  3814 I ActivityTaskManager: START u0 {act=com.google.android.gms.signin.action.SIGN_IN cmp=com.google.android.gms/.signin.activity.SignInActivity (has extras)} from uid 10044
02-07 21:53:36.238   977  3814 E Parcel  : Class not found when unmarshalling: com.google.android.gms.common.api.Scope
02-07 21:53:36.238   977  3814 E Parcel  : java.lang.ClassNotFoundException: com.google.android.gms.common.api.Scope
02-07 21:53:36.238   977  3814 E Parcel  : 	at java.lang.Class.classForName(Native Method)
02-07 21:53:36.238   977  3814 E Parcel  : 	at java.lang.Class.forName(Class.java:454)
```

**Comment by nabati** [comment]
#120 seems related, but has been left unresolved.

**Comment by nabati** [comment]
Above exception seems unrelated to the oauth consent screen being stuck in loading.

Installing the app in an emulator instead of on device doesn't render above error, but still the consent screen is stuck loading. Some googling indicates above error is related to Samsung devices specifically.

![image](https://user-images.githubusercontent.com/1499050/107186864-576b2900-69e5-11eb-9628-f89040ed0957.png)

I have confirmed that the SHA1-fingerprint of the apk installed is correctly added to the Google API Console Oauth Client ID, and that the package name is correct. 

For future travellers, you can find the SHA1-fingerprint of your apk via; 

(for debug)
```
keytool -printcert -jarfile android/app/build/outputs/apk/debug/app-debug.apk
```

To make sure the package name is correct, you can also check the apk (for debug)

```
aapt dump badging android/app/build/outputs/apk/debug/app-debug.apk | grep package
```

**Comment by nabati** [comment]
Attempted using the same keystore and package name using the official Google Fit [sample](https://github.com/android/fit-samples/tree/master/BasicHistoryApiKotlin) and get the same error, so most likely something is misconfigured.

**Comment by nabati** [comment]
Found a solution. 

Seems like there's a bug on Google's side? 

My consent screen was in "Testing" mode, which reading this text should allow me to test it (I've added the relevant emails as test users). 

![image](https://user-images.githubusercontent.com/1499050/107190946-c8154400-69eb-11eb-9ba7-2ddd27972981.png)

If you press "Publish app", then it will work.

![image](https://user-images.githubusercontent.com/1499050/107190885-addb6600-69eb-11eb-9c04-4c768d81ac2c.png)

Related link; 
https://issuetracker.google.com/issues/178183308


🪦 time spent
